data/lua/core/unit_test.lua

```diff
@@ -0,0 +1,146 @@
+
+--[========[Unit Test Helpers]========]
+
+if rawget(_G, 'unit_test') ~= nil then
+    print("Loading unit_test module...")
+
+	function unit_test.finish(result)
+		wesnoth.wml_actions.endlevel{
+			test_result = result,
+			linger_mode = true
+		}
+	end
+
+	function unit_test.fail()
+		unit_test.finish('fail')
+	end
+
+	function unit_test.pass()
+		unit_test.finish('pass')
+	end
+
+    function unit_test.tostring(val)
+        -- This exists so custom behaviour can be added for specific types if required.
+		if type(val) == 'string' then
+			-- Strings are output quoted but without escaped_split
+			-- Pick a quote style that the string doesn't contain
+			if val:find("'") == nil then
+				return "'" .. val .. "'"
+			elseif val:find('"') == nil then
+				return '"' .. val .. '"'
+			else
+				local n = 0;
+				-- This loop is guaranteed to end eventually, although it could run a very large number of times on particularly pathological input.
+				while true do
+					local left = '[' .. string.rep('=', n) .. '['
+					local right = ']' .. string.rep('=', n) .. ']'
+					if val:find(left) == nil and val:find(right) == nil then
+						return left .. val .. right
+					end
+					n = n + 1
+				end
+			end
+		end
+        return tostring(val)
+    end
+
+    function unit_test.log(prefix, message)
+        std_print(prefix .. ': ' .. message)
+    end
```

Comment:
Inconsistent indenting for log and tostring.

Reply:
Actually just a lot of weird indentation in this file overall.

Reply:
I've fixed this indentation in #5708